shell/platform/windows/platform_view_manager.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/platform_view_manager.h"
+
+#include "flutter/shell/platform/common/client_wrapper/include/flutter/standard_method_codec.h"
+
+namespace flutter {
+
+namespace {
+constexpr char kChannelName[] = "flutter/platform_views";
+}
+
+PlatformViewManager::PlatformViewManager(TaskRunner* task_runner,
+                                         BinaryMessenger* binary_messenger)
+    : task_runner_(task_runner),
+      channel_(std::make_unique<MethodChannel<EncodableValue>>(
+          binary_messenger,
+          kChannelName,
+          &StandardMethodCodec::GetInstance())) {
+  channel_->SetMethodCallHandler(
+      [this](const MethodCall<EncodableValue>& call,
+             std::unique_ptr<MethodResult<EncodableValue>> result) {
+        const auto& args = std::get<EncodableMap>(*call.arguments());
+        if (call.method_name() == "create") {
+          const auto& type =
+              std::get<std::string>(args.find(EncodableValue("type"))->second);
+          const auto& id =
+              std::get<std::int32_t>(args.find(EncodableValue("id"))->second);
+          QueuePlatformViewCreation(type, id);
+        }
+        result->Success();
+      });
```

Comment:
I know this isn't a pattern the Windows embedder uses today, but let's separate messaging logic from implementation logic similar to how Pigeon does it.

Example from the `file_selector_windows` plugin:

1. [`FileSelectorApi`](https://github.com/flutter/packages/blob/abb04bb89cca97e653764a60bde0ee4e816db322/packages/file_selector/file_selector_windows/windows/messages.g.h#L162-L186) - This is the base type that defines the plugin's full API surface
    1. It defines some `virtual` methods that the child implementation type needs to implement: [link](https://github.com/flutter/packages/blob/abb04bb89cca97e653764a60bde0ee4e816db322/packages/file_selector/file_selector_windows/windows/messages.g.h#L167-L173)
	1. [`SetUp` method](https://github.com/flutter/packages/blob/abb04bb89cca97e653764a60bde0ee4e816db322/packages/file_selector/file_selector_windows/windows/messages.g.cpp#L208-L296) - Registers the handlers.
1. [`FileSelectorPlugin`](https://github.com/flutter/packages/blob/abb04bb89cca97e653764a60bde0ee4e816db322/packages/file_selector/file_selector_windows/windows/file_selector_plugin.h#L22) - This is the child type that handles all the implementation logic.

Reply:
I am still somewhat unsure what pattern you are hinting at, or how it would apply here. Do you think you can elaborate on what you mean?

Reply:
I think this is done, marking as resolved!